src/pybind/mgr/cephadm/module.py

```diff
@@ -979,6 +979,9 @@ def set_ssh_user(self, user: str) -> Tuple[int, str, str]:
             return 0, "value unchanged", ""
 
         self._validate_and_set_ssh_val('ssh_user', user, current_user)
```

Comment:
Isn't this line supposed to set the **ssh_user**? this code sets the new value and then calls: `self.ssh._reconfig_ssh()` which does the following:

```
....
        if self.mgr.mode == 'root':
            self.mgr.ssh_user = self.mgr.get_store('ssh_user', default='root')
        elif self.mgr.mode == 'cephadm-package':
            self.mgr.ssh_user = 'cephadm'
....
```
so it seems that in order to change the ssh_user `self.mgr.mode` must be **root**

Reply:
Yes it sets the ssh_user kv but set-user should also change the value in ssh-config:

> Host *
>   User root    <------------
>   StrictHostKeyChecking no
>   UserKnownHostsFile /dev/null
>   ConnectTimeout=30

This is the changes I added. I may be missing something here...

Reply:
To give a bit of info on the root vs. package mode thing, way back ther ewas an idea to allow users to just have a cephadm package installed on the host that we would use for all cephadm commands. While the code for it still exists (https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/serve.py#L1328) it's never properly worked and has been effectively deprecated for 2 years or so. That beings said, it would still be easy enough to respect it here. We would just need to only change the config like this in cases where `if self.mode == 'root'` passes. Although, since package mode isn't used at all afaik and is mostly likely just broken currently it really wouldn't make a tangible difference.